ToolBox/QuiverDataDownloader/QuiverSenateDataDownloader.cs

```diff
@@ -0,0 +1,153 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using QuantConnect.Configuration;
+using QuantConnect.Data.Auxiliary;
+using QuantConnect.Data.Custom.Quiver;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QuantConnect.ToolBox.QuiverDataDownloader
+{
+
+    public class QuiverSenateDataDownloader : QuiverDataDownloader
+    {
+        private readonly string _destinationFolder;
+        private readonly MapFileResolver _mapFileResolver;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="QuiverSenateDataDownloader"/>
+        /// </summary>
+        /// <param name="destinationFolder">The folder where the data will be saved</param>
+        public QuiverSenateDataDownloader(string destinationFolder)
+        {
+            _destinationFolder = Path.Combine(destinationFolder, "senatetrading");
+            _mapFileResolver = Composer.Instance.GetExportedValueByTypeName<IMapFileProvider>(Config.Get("map-file-provider", "LocalDiskMapFileProvider"))
```

Comment:
nit: `_mapFileResolver` is unused across your data downloader/converters. Thankfully, you store data in a way that doesn't require the use of it